R/zzz.R

```diff
@@ -18,3 +18,21 @@
 
 
 ### END GLOBAL VARIABLES ###
+
+
+### ENSURE CHECK PASSES
+
+# this function is necessary for check to properly process code dependencies within R6 classes
+.rectify_dependencies_check <- function() {
+  dplyr::filter
+  grDevices::rgb
+  lifecycle::badge
+  logger::log_trace
+  plotly::plot_ly
+  shinycssloaders::withSpinner
+  shinyWidgets::pickerOptions
+  teal.widgets::optionalSelectInput
+}
```

Comment:
I don't understand this. Can you explain?
Do I understand properly that checks doesn't recognise usage of the functions in methods of R6 classes?

Reply:
That is exactly what happens.

If `package` is listed in `Imports`, (1) check looks through `NAMESPACE` to see if there are any `importFrom(package,<foo>)` statements or an `import(package)` statement. If none are found, (2) check looks for `package::*` calls in the code. If none are found again, (3) check throws the NOTE, which is banned by our CI.
Now, when `package::*` statements are made within an `R6` class, they are not registered. Since `plotly` and `shinycssloaders` were only used within `RangeFilterState$server_inputs`, I kept getting 
```
  Namespaces in Imports field not imported from:
    'plotly' 'shinycssloaders'
    All declared Imports should be used.
```